guide.py

```diff
@@ -0,0 +1,434 @@
+from __future__ import division
+from __future__ import print_function
+
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+import json
+import logging
+import os
+import re
+import sys
+import webapp2
+from bs4 import BeautifulSoup
+from HTMLParser import HTMLParser
+from xml.dom import minidom
+from django import forms
+
+# Appengine imports.
+from google.appengine.api import memcache
+from google.appengine.api import users
+from google.appengine.ext import db
+from google.appengine.api import taskqueue
+
+# File imports.
+import common
+import guideforms
+import models
+import processes
+import settings
+
+
+# Forms to be used for each stage of the process.
+STAGE_FORMS = {
+    models.INTENT_NONE: guideforms.Incubate,
+    models.INTENT_IMPLEMENT: guideforms.Prototype,
+    models.INTENT_EXPERIMENT: guideforms.DevTrial,
+    models.INTENT_EXTEND_TRIAL: guideforms.OriginTrial,
+}
+
+class FeatureNew(common.ContentHandler):
+
+  def get(self, path):
+    user = users.get_current_user()
+    if user is None:
+      return self.redirect(users.create_login_url(self.request.uri))
+
+    # TODO(ericbidelman): This creates a additional call to
```

Comment:
There are lots of TODO's in the code pinned to ericbidelman.  If we want to retain them, they should get updated to reflect a current owner.

Are there opportunities for code reuse to improve maintainability?

Reply:
I deleted these two TODOs because I don't think the they are helpful.  But, as I touch other code, I will update the TODO owner as I go.